### ISSUE TITLE
How to Calculate grid features from given pretrained X152pp model

### ISSUE THREAD

**Issue by ifmaq1** [facebookresearch/grid-feats-vqa#34]
I am trying to extract the features for test split of Coco dataset. I used the pretrained model of Backbone X-152++ 

to extract features I used
python3.6 extract_grid_feature.py --config-file configs/X-152-challenge.yaml --dataset coco_2015_test MODEL.WEIGHTS /home/grid-feats-vqa-master/output/X-152pp.pth

However, when I matched the calculated features with https://github.com/facebookresearch/mmf features for **movie_mcan** model. **I found a huge file size gap between this repository features and MMF's**

i-e

The test split image features for X-152pp

The one I calculated myself is of size
3.9M 1024.pth

The one I downloaded from **mmf** for Movie_MCAN is of size
8.1M 1024.pth

the features that I calculated for the whole **test split** of **vqa2** for **X-152pp** are of size approx. 320gb 
please let me know how there could be huge file size difference

**Comment by ifmaq1** [comment]
Here are some initial logs upon writing a command

**python3.6 extract_grid_feature.py --config-file configs/X-152-challenge.yaml --dataset coco_2015_test MODEL.WEIGHTS /home/grid-feats-vqa-master/output/X-152pp.pth**



Command Line Args: Namespace(config_file='configs/X-152-grid.yaml', dataset='coco_2015_test', opts=['MODEL.WEIGHTS', '/home/grid-feats-vqa-master/output/X-152pp.pth'])
[05/07 22:45:37 detectron2]: Rank of current process: 0. World size: 1
[05/07 22:45:45 detectron2]: Environment info:
------------------------  ---------------------------------------------------------------
sys.platform              linux
Python                    3.6.12 (default, Aug 18 2020, 02:08:22) [GCC 5.4.0 20160609]
numpy                     1.17.4
detectron2                0.1.1 @/usr/local/lib/python3.6/dist-packages/detectron2
detectron2 compiler       GCC 5.5
detectron2 CUDA compiler  8.0
detectron2 arch flags     sm_37
DETECTRON2_ENV_MODULE     <not set>
PyTorch                   1.4.0+cu100 @/usr/local/lib/python3.6/dist-packages/torch
PyTorch debug build       False
CUDA available            True
GPU 0                     Tesla K80
CUDA_HOME                 /usr/local/cuda
NVCC                      Cuda compilation tools, release 10.1, V10.1.243
Pillow                    8.0.1
torchvision               0.5.0+cu100 @/usr/local/lib/python3.6/dist-packages/torchvision
torchvision arch flags    sm_35, sm_50, sm_60, sm_70, sm_75
------------------------  ---------------------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - Intel(R) Math Kernel Library Version 2019.0.4 Product Build 20190411 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CUDA Runtime 10.0
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.1
  - Build settings: BLAS=MKL, BUILD_NAMEDTENSOR=OFF, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Wno-stringop-overflow, DISABLE_NUMA=1, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF,

[05/07 22:45:45 detectron2]: Command line arguments: Namespace(config_file='configs/X-152-grid.yaml', dataset='coco_2015_test', opts=['MODEL.WEIGHTS', '/home/grid-feats-vqa-master/output/X-152pp.pth'])
[05/07 22:45:45 detectron2]: Contents of args.config_file=configs/X-152-grid.yaml:
_BASE_: "Base-RCNN-grid.yaml"
MODEL:
  WEIGHTS: "catalog://ImageNetPretrained/FAIR/X-152-32x8d-IN5k"
  RESNETS:
    STRIDE_IN_1X1: False  # this is a C2 model
    NUM_GROUPS: 32
    WIDTH_PER_GROUP: 8
    DEPTH: 152
[05/07 22:45:45 detectron2]: Running with full config:
CUDNN_BENCHMARK: False
DATALOADER:
  ASPECT_RATIO_GROUPING: True
  FILTER_EMPTY_ANNOTATIONS: True
  NUM_WORKERS: 4
  REPEAT_THRESHOLD: 0.0
  SAMPLER_TRAIN: TrainingSampler
DATASETS:
  PRECOMPUTED_PROPOSAL_TOPK_TEST: 1000
  PRECOMPUTED_PROPOSAL_TOPK_TRAIN: 2000
  PROPOSAL_FILES_TEST: ()
  PROPOSAL_FILES_TRAIN: ()
  TEST: ('visual_genome_test',)
  TRAIN: ('visual_genome_train', 'visual_genome_val')
GLOBAL:
  HACK: 1.0
INPUT:
  CROP:
    ENABLED: False
    SIZE: [0.9, 0.9]

**Comment by endernewton** [comment]
Hmm, I am not sure. Is it because of padding?

**Comment by ifmaq1** [comment]
padding in MMF features? 
Also due to this there is a huge accuracy difference on MOVIE_MCAN model. Can you please somehow tell where the problem lies. Maybe confirming from https://github.com/facebookresearch/mmf!!
